File: src/classes/classes.pl
#!/usr/bin/env perl

package Dog {

  sub new {
    my ( $class, $name, $breed ) = @_;
    my $self = {
      breed => $breed,
      name  => $name,
    };

    bless $self, $class;
    return $self;
  }

  sub bark {
    my $self = shift;
    print "ruff! ruff!\n";
  }

  sub tag {
    my $self = shift;
    return $self->name . " the " . $self->breed;
  }
}

my $dog = Dog->new("Finn", "Labrador");

# paranthesis are optional
print $dog->tag() . "\n";
$dog->bark();

File: src/classes/classes.py
#!/usr/bin/env python

class Dog(object):

    def __init__(self, name, breed):
        self.name = name
        self.breed = breed

    def bark(self):
        print "ruff! ruff!"

    def tag(self):
        return self.name + " the " + self.breed


dog = Dog("Finn", "Labrador")

# paranthesis are required
print dog.tag()
dog.bark()
